### ISSUE TITLE
How to add python stacks to the predefined-stack.json

### ISSUE THREAD

**Issue by gma-git-admin** [eclipse-che/che#1129]
Hi,

Am new to Che and having difficulty trying to find my way here. I want to use the python images in codenvy dockerhub and add that to predefined_stack.json file. So that python stack is available via dashboard page in Che.

When I do that however, it doesn't work with status looking like this:
[DOCKER] Step1: FROM codenvy/python27
[DOCKER] ---> 35176b922f1c
[DOCKER] ---> Successfully built 35176b922f1c
Machine is destroyed.

Am not able to get hang of it. Please help.
By the way I could create Python workspace by using Docker file and selecting Create New Workspace from Custom Stack->Write your own stack.

**Comment by ghost** [comment]
@gma-git-admin ,
`codenvy/ubuntu_python` is Python 3.5 while `codenvy/ubuntu_python:2.7` is 2.7 version

Btw, these two python stacks are in master already https://github.com/eclipse/che/releases/tag/4.2.0 so, updating your Che will fix the problem.

Otherwise, you are on the right track to start workspace from custom Dockerfiles.

**Comment by gma-git-admin** [comment]
Using the ones from codenvy DockerHub  doesn't work but it does work using docker files from https://github.com/codenvy/dockerfiles/blob/master/ubuntu_python/3.5/Dockerfile. 

To add that to stack library, I followed these steps:
Create the workspace using dockerfile.
Identified the container created from Docker 
Create a Docker image out of this new container.
Add a new entry in predefined_stacks.json for python using this new image.

And it worked! Though I guessed new version have Python stack already built in. Will update that. Also couldn't understand the recipe bit. 

Now am fiddling with linking this to Git repo. I couldn't follow the steps documented here https://eclipse-che.readme.io/docs/git. Am using public repository and not private. 

And finally, thanks **@eivantsov** for prompt replies it makes a big difference! Che seems to be game changer for cloud-based IDE and we are looking forward to transition to it.
